### ISSUE TITLE
Inline tests results

### ISSUE THREAD

**Issue by yamsellem** [TabDigital/atom-mocha-test-runner#45]
Looking at [atom wallaby support](https://github.com/wallabyjs/public/issues/24#issuecomment-123173447), having tests results in the margin is quite impressive.
Is this doable with mocha and the atom test runner?

![unnamed](https://cloud.githubusercontent.com/assets/606754/8797696/757edb7c-2f9d-11e5-8567-7dc02393ded3.gif)

**Comment by rprieto** [comment]
I have to say that's pretty cool! The whole variable evaluation is probably out of scope for `mocha-test-runner`, but marking the test definition as red might be possible... and same for marking the line that threw the exception. Let's see what @nguyenchr thinks.

**Comment by kiddkai** [comment]
@rprieto , it's super cool!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! I will come back to `atom` if this implemented.

**Comment by nguyenchr** [comment]
this is very cool, I agree that we could probably do the test definition and the line that threw the exception

**Comment by rprieto** [comment]
Now that I remember though, we closed a similar issue a while ago (#22).
I guess it comes down to how much of `wallaby` we want to emulate - without becoming wallaby itself :)

**Comment by yamsellem** [comment]
Wallaby is a closed source all or nothing toolset combining mocha + blanket + a runner (and heavily inspired by those). 

The code coverage part does not seem to be relevant here. But afaik, It is possible to create a better open source test runner. And this runner won't be anything like wallaby because it will be based on open source, hackable by the community, and be one little tool of a larger toolset. 

Someone may even fork it for karma. 
Keeping the unix "small tool for simple task" philosophy seems right too me ;-).

**Comment by rprieto** [comment]
Fair enough about wallaby being close-sourced. I have a different take on `small tool for simple task` though :smile: For now, `mocha-test-runner` runs tests on demand, and display the full test output. From what I see in the screenshot, wallaby runs tests continuously and displays red/green in the sidebar. Should this be a different Atom plugin then? 
- `mocha-test-runner` for running on demand (current file or selected test), and display full results
- `mocha-gutter` for running *all* tests continuously, and updating the colors in the gutter

I could be convinced either way, but it's a big feature change for this plugin so would like to hear some more opinions.

**Comment by yamsellem** [comment]
You're right, there're several things here:
- displaying the results in the gutter 
- running the test continuously 
- linking the test and the required files, so, on change of one of these, rerun the test (which may be the coolest but hardest feature)

You may or may not see those features appropriate for the mocha runner. To me, they serve a same purpose, and because I've already used and loved tools like this in other languages, I'd be super happy to have something similar in JavaScript. 

May I suggest splitting this issue into its three parts and start thinking of each question independently?

**Comment by kiddkai** [comment]
then the modules are

- mocha-gutter
- mocha-gutter-istanbul
- mocha-gutter-blancket

?

**Comment by yamsellem** [comment]
I'm not sure this has to be split that way. Either scenario, it may look like this:
- `mocha-gutter`, displaying mocha tests results in the gutter
- `mocha-?`, running the test continuously
- `mocha-?`, linking the test and the required files, so, on change of one of these, rerun the test (which may be the coolest but hardest feature)
- `mocha-gutter-istanbul` & `mocha-gutter-blancket`, displaying coverage in the source

**Comment by yamsellem** [comment]
There has been a lot of water under the bridge since our last debate on this.
Have you seen any improvement in the open source field on the matter?

**Comment by SpoBo** [comment]
https://atom.io/packages/test-gutter perhaps create a plugin that works with this? The author made one runner that works with rspec. It's probably a matter of parsing mocha output and giving it to the test-gutter.
